pdg/src/snapshots/c2rust_pdg__tests__analysis_test_pdg_snapshot_debug.snap

```diff
@@ -3,36 +3,36 @@ source: pdg/src/main.rs
 expression: pdg
 ---
 g {
-	n[0]: &_1  _    => _11 @ bb3[9]: fn main;   _11 = &_1;
-	n[1]: copy n[0] => _1  @ bb0[0]: fn deref;  _10 = deref(move _11);
+	n[0]: &_1  _    => _   @ bb3[0]:  fn main;   _23 = &raw mut _1;
+	n[1]: copy n[0] => _11 @ bb3[10]: fn main;   _11 = &(*_23);
+	n[2]: copy n[1] => _1  @ bb0[0]:  fn deref;  _10 = deref(move _11);
 }
 nodes_that_need_write = []
 
 g {
-	n[0]: copy _    => _10 @ bb3[10]: fn main;  _10 = deref(move _11);
+	n[0]: copy _    => _10 @ bb3[11]: fn main;  _10 = deref(move _11);
 	n[1]: copy n[0] => _9  @ bb4[0]:  fn main;  _9 = &(*_10);
 	n[2]: copy n[1] => _1  @ bb0[0]:  fn iter;  _8 = iter(move _9);
 }
 nodes_that_need_write = []
 
 g {
-	n[0]: copy        _    => _14   @ bb6[4]: fn main;                _14 = null_mut();
-	n[1]: copy        n[0] => _1    @ bb0[0]: fn once;                _13 = once(move _14);
-	n[2]: value.store _    => _20.* @ bb4[7]: fn invalid;             (*_20) = const 0_usize as *mut pointers::S (PointerFromExposedAddress);
-	n[3]: value.store _    => _17.* @ bb8[4]: fn fdevent_unregister;  (*_17) = const 0_usize as *mut pointers::fdnode_st (PointerFromExposedAddress);
-	n[4]: int_to_ptr  _    => _2    @ bb0[2]: fn test_ref_field;      _2 = const 0_usize as *const pointers::S (PointerFromExposedAddress);
-	n[5]: int_to_ptr  _    => _5    @ bb0[8]: fn test_ref_field;      _5 = const 0_usize as *const pointers::S (PointerFromExposedAddress);
+	n[0]: copy        _    => _14    @ bb6[4]:  fn main;                 _14 = null_mut();
+	n[1]: copy        n[0] => _1     @ bb0[0]:  fn once;                 _13 = once(move _14);
+	n[2]: value.store _    => _20.*  @ bb4[7]:  fn invalid;              (*_20) = const 0_usize as *mut pointers::S (PointerFromExposedAddress);
+	n[3]: value.store _    => _17.*  @ bb8[4]:  fn fdevent_unregister;   (*_17) = const 0_usize as *mut pointers::fdnode_st (PointerFromExposedAddress);
+	n[4]: int_to_ptr  _    => _2     @ bb0[2]:  fn test_ref_field;       _2 = const 0_usize as *const pointers::S (PointerFromExposedAddress);
+	n[5]: int_to_ptr  _    => _5     @ bb0[8]:  fn test_ref_field;       _5 = const 0_usize as *const pointers::S (PointerFromExposedAddress);
+	n[6]: int_to_ptr  _    => _46    @ bb34[3]: fn main_0;               _46 = const 0_usize as *const pointers::S (PointerFromExposedAddress);
+	n[7]: value.store _    => _3.*.2 @ bb0[1]:  fn test_addr_taken_arg;  ((*_3).2: *const pointers::S) = const 0_usize as *const pointers::S (PointerFromExposedAddress);
```

Comment:
I guess this is the graph for the null pointer?  Seems kind of weird to have such a thing - maybe it should be filtered out.  But any such changes can be left to a separate PR (and we should probably think more about `IntToPtr` as part of that).